.github/workflows/ci.yml

```
@@ -0,0 +1,102 @@
+name: ci
+on:
+  push:
+    branches:
+    - master
+    tags:
+    - "v[0-9]+.[0-9]+.[0-9]+"
+  pull_request:
+
+permissions:
+  contents: read
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+    - name: ci/checkout-repo
+      uses: actions/checkout@755da8c3cf115ac066823e79a1e1788f8940201b # v3.2.0
+      with:
+        fetch-depth: 0
+
+    - name: ci/setup-go
+      uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # v3.5.0
+      with:
+        go-version: "1.19"
```

You can use global env here
```yaml
env:
  GO_VERSION: 1.19
```
and reuse when needed
```suggestion
        go-version: ${{ env.GO_VERSION }}
```